app/controllers/evaluations_controller.rb

```diff
@@ -50,38 +55,61 @@ def new
   end
 
   def edit
-    @evaluation = Evaluation.find_by(id: params[:id])
+    @evaluation = Evaluation.includes([evaluation_scores: :evaluation_criterion]).find_by(id: params[:id])
     @evaluator_submission_assignment = find_evaluator_submission_assignment
     return unauthorized_redirect unless can_access_evaluation?
 
     render :edit
   end
 
-  def save_draft
-    @evaluation.completed_at = nil
+  def create
+    if save_evaluation
+      flash[:notice] =
+        if params[:subaction] == "mark_complete"
+          I18n.t("evaluations.notices.marked_complete")
+        else
+          I18n.t("evaluations.notices.saved_draft")
+        end
 
-    if @evaluation.save(validate: false)
-      flash[:notice] = I18n.t("evaluations.notices.saved_draft")
-      redirect_to evaluations_path
+      redirect_to submissions_evaluation_path(@evaluation.submission)
     else
-      handle_save_draft_failure
+      render :new, status: :unprocessable_entity
     end
   end
 
-  def mark_complete
-    @evaluation.completed_at = Time.current
+  def update
+    if save_evaluation
+      flash[:notice] =
+        if params[:subaction] == "mark_complete"
+          I18n.t("evaluations.notices.marked_complete")
+        else
+          I18n.t("evaluations.notices.saved_draft")
+        end
 
-    if @evaluation.save
-      flash[:notice] = I18n.t("evaluations.notices.marked_complete")
-      redirect_to evaluations_path
+      redirect_to submissions_evaluation_path(@evaluation.submission)
```

Comment:
same as above, url is incorrect
```suggestion
      redirect_to submissions_evaluation_path(@evaluation.submission.phase)
```